cl/search/management/commands/pacer_bulk_fetch.py

```diff
@@ -0,0 +1,385 @@
+from cl.scrapers.tasks import extract_recap_pdf
+from cl.search.models import Court, RECAPDocument
+
+logger = logging.getLogger(__name__)
+
+
+def append_value_in_cache(key, value):
+    cached_docs = cache.get(key)
+    if cached_docs is None:
+        cached_docs = []
+    cached_docs.append(value)
+    one_month = 60 * 60 * 24 * 7 * 4
+    cache.set(key, cached_docs, timeout=one_month)
+
+
+class Command(VerboseCommand):
+    help = "Download multiple documents from PACER with rate limiting"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.options = None
+        self.user = None
+        self.recap_documents = None
+        self.courts_with_docs = {}
+        self.total_launched = 0
+        self.total_errors = 0
+        self.max_retries = 5
+        self.pacer_username = None
+        self.pacer_password = None
+        self.throttle = None
+        self.queue_name = None
+        self.interval = None
+        self.fetches_in_progress = {}  # {court_id: (fq_pk, retry_count)}
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "--interval",
+            type=float,
+            help="The minimum wait in secs between PACER fetches to the same court.",
+        )
+        parser.add_argument(
+            "--min-page-count",
+            type=int,
+            help="Get docs with this number of pages or more",
+        )
+        parser.add_argument(
+            "--max-page-count",
+            type=int,
+            help="Get docs with this number of pages or less",
+        )
+        parser.add_argument(
+            "--username",
+            type=str,
+            help="Username to associate with the processing queues (defaults to 'recap')",
+        )
+        parser.add_argument(
+            "--queue-name",
+            type=str,
+            help="Celery queue name used for processing tasks",
+        )
+        parser.add_argument(
+            "--testing",
+            type=str,
+            help="Prevents creation of log file",
+        )
+        parser.add_argument(
+            "--stage",
+            type=str,
+            choices=["fetch", "process"],
+            default="fetch",
+            help="Stage of the command to run: fetch or process",
+        )
+
+    @staticmethod
+    def docs_to_process_cache_key():
+        """Helper method to improve testability."""
+        return "pacer_bulk_fetch.docs_to_process"
+
+    @staticmethod
+    def timed_out_docs_cache_key():
+        """Helper method to improve testability."""
+        return "pacer_bulk_fetch.timed_out_docs"
+
+    @staticmethod
+    def setup_logging(testing: bool = False) -> None:
+        if not testing:
+            logging.basicConfig(
+                filename=f'pacer_bulk_fetch_{datetime.now().strftime("%Y%m%d_%H%M%S")}.log',
+                level=logging.INFO,
+                format="%(asctime)s - %(levelname)s - %(message)s",
+            )
```

Comment:
Is there a reason for collecting logs into a file for this command? We usually monitor logs via Loki in Grafana, so this might not be necessary?

Reply:
Removed this method. Since we don't require to store logs in a file.